titiler/pgstac/factory.py

```diff
@@ -699,6 +718,166 @@ def info_search(request: Request, searchid=Depends(self.path_dependency)):
                 ],
             )
 
+    def _search_list_routes(self) -> None:
+        """Add mosaic listing route."""
+
+        @self.router.get(
+            "/list",
+            responses={200: {"description": "List Mosaics in PgSTAC."}},
+            response_model=model.Infos,
+            response_model_exclude_none=True,
+        )
+        def list_mosaic(
+            request: Request,
+            limit: int = Query(
+                10,
+                ge=1,
+                le=1000,
+                description="Page size limit",
+            ),
+            offset: int = Query(
+                0,
+                ge=0,
+                description="Page offset",
+            ),
+            sortby: Optional[str] = Query(
+                None,
+                description="Sort the response items by a property (ascending (default) or descending).",
+            ),
+        ):
+            """List a Search query."""
+            # Default filter to only return `metadata->type == 'mosaic'`
+            mosaic_filter = sql.SQL("metadata->>'type' = 'mosaic'")
+
+            # additional metadata property filter passed in query-parameters
+            # <propname>=val - filter for a metadata property. Multiple property filters are ANDed together.
+            qs_key_to_remove = ["limit", "offset", "sortby"]
+            additional_filter = [
+                sql.SQL("metadata->>{key} = {value}").format(
+                    key=sql.Literal(key), value=sql.Literal(value)
+                )
+                for (key, value) in request.query_params.items()
+                if key.lower() not in qs_key_to_remove
+            ]
+            filters = [
+                sql.SQL("WHERE"),
+                sql.SQL("AND ").join([mosaic_filter, *additional_filter]),
+            ]
+
+            def parse_sort_by(sortby: str) -> Generator[sql.Composable, None, None]:
+                """Parse SortBy expression."""
+                for s in sortby.split(","):
+                    parts = re.match(
+                        "^(?P<dir>[+-]?)(?P<prop>.*)$", s.lstrip()
```

Comment:
I add to add `s.lstrip()` because I tried to pass `sortby=+{prop}` I couldn't encode it properly and it resulted on `sortby= {prop}` which because `+` is the defaults gives the same results